src/Interpreters/Streaming/SubstituteStreamingFunction.h

```diff
@@ -28,6 +28,11 @@ struct StreamingFunctionData
 
     /// only streaming query can use these functions
     static std::set<String> streaming_only_func;
+
+    /// check whether or not the function support 'retract' for changelog:
+    /// the 1st return value: indicates whether or not it support retract
+    /// the 2nd return value: the alias name of function should be rewritten in the query, empty if not support changelog
+    std::pair<bool, String> supportChangelog(const String & function_name);
```

Comment:
`supportsChangelog(...)`

Reply:
I think a `std::optional<String> supportsChange(...)` signature works better ? If the target func supports retract a non-empty std::optional is returned and the the function alias name which shall be rewritten. Otherwise an empty std::optional is returned